src/Features/LanguageServer/ProtocolUnitTests/Ordering/RequestOrderingTests.cs

```diff
@@ -213,9 +213,21 @@ public async Task NonMutatingRequestsOperateOnTheSameSolutionAfterMutation()
 
             expectedSolution = testLspServer.GetCurrentSolution();
 
-            // solution should be different because there has been a workspace change
             solution = await GetLSPSolution(testLspServer, NonMutatingRequestHandler.MethodName);
-            Assert.NotEqual(expectedSolution, solution);
+
+            if (mutatingLspWorkspace)
+            {
+                // In the case where this is a mutating workspace, we would expect the solutions to be the same as
+                // everything pushes through between the manager and the workspace, and there have been no text changes
+                // in one to cause us to fork.
+                Assert.Equal(expectedSolution, solution);
```

Comment:
I admit I'm not getting this test being a theory in the way it is: the comment on line 198 said we forked. Should this just be rewritten to two tests, one going either way?